src/biokbase/narrative/tests/test_appmanager.py

```diff
@@ -365,6 +366,71 @@ def test_run_app_batch_missing_inputs(self, auth, c):
 
     ############# End tests for run_app_batch #############
 
+    ############# Test run_local_app #############
+    @mock.patch("biokbase.narrative.jobs.appmanager.clients.get", get_mock_client)
+    @mock.patch("biokbase.narrative.jobs.appmanager.JobComm")
+    @mock.patch(
+        "biokbase.narrative.jobs.appmanager.auth.get_agent_token",
+        side_effect=mock_agent_token,
+    )
+    def test_run_local_app_ok(self, auth, c):
+        c.return_value.send_comm_message = MagicMock()
+        result = self.am.run_local_app(
+            self.test_viewer_app_id,
+            {"param0": "fakegenome"},
+            tag="release",
+        )
+        self.assertIsInstance(result, Javascript)
+        self.assertIn("KBaseNarrativeOutputCell", result.data)
+
+    @mock.patch("biokbase.narrative.jobs.appmanager.clients.get", get_mock_client)
+    @mock.patch("biokbase.narrative.jobs.appmanager.JobComm")
+    @mock.patch(
+        "biokbase.narrative.jobs.appmanager.auth.get_agent_token",
+        side_effect=mock_agent_token,
+    )
+    def test_run_local_app_fail_cases(self, auth, c):
+        comm_mock = MagicMock()
+        c.return_value.send_comm_message = comm_mock
+        cases = [{
+            "inputs": {
+                "args": [self.bad_app_id, {}],
+                "kwargs": {}
+            },
+            "expected_error": f'Unknown app id "{self.bad_app_id}" tagged as "release"',
+        }, {
+            "inputs": {
+                "args": [self.test_viewer_app_id, {}],
+                "kwargs": {"tag": "dev"}
+            },
+            "expected_error": "Missing required parameters"
+        }, {
+            "inputs": {
+                "args": [self.good_app_id, {}],
+                "kwargs": {"tag": self.bad_tag}
+            },
+            "expected_error": f"Can't find tag {self.bad_tag} - allowed tags are release, beta, dev"
+        }, {
+            "inputs": {
+                "args": [self.good_app_id, {}],
+                "kwargs": {"tag": "dev", "version": "1.0.0"}
+            },
+            "expected_error": "Semantic versions only apply to released app modules.",
```

Comment:
Sonarcloud points out that this string has been repeated three times -- may be worth turning it into a constant and using that in the tests.